src/exoscale/compute/api/payload.clj

```diff
@@ -36,7 +36,7 @@
         (transform-maps (name k) v)
 
         (sequential? v)
-        [[k (str/join "," (map name v))]]
+        (map-indexed #([k (if (keyword? %) (name %) (str %))]) v)
```

Comment:
Maybe I'm reading this wrong but `map-indexed` expects a function of two values. Was the argument previously bad formatted, since this also changes the way things will be sent. I think the sanitization below suggests this code path is not hit anymore, I would thus revert to the previous behavior.

Reply:
Good catch! My bad.

Indeed, this code path is not hit for our particular case at the moment because I have added the sanitisation before. Still, the previous behaviour is wrong and we could hit it again depending on the case.

The problem is that Aleph serialize sequences differently. If you pass for instance `{:seq [1 2]}`, Aleph will spit out something like `seq=1&seq=2` whereas we generate `seq=1,2`. In fine, Aleph and our lib generate different payload which leads to different signatures. See here: https://github.com/ztellman/aleph/blob/master/src/aleph/http/client_middleware.clj#L410

I think we should stick to the same implementation than Aleph for this to avoid loosing time the next time (lost ~4 hours to catch this one). I will fix the code though ;)

OK for you?

PS: We really need to write tests for the lib now.

Reply:
What aleph does is how multiple instances of the same query args are supposed to function. But for the cloudstack compatibility API, lists need to be comma separated strings, the input param handler does not know how to handle those. Since this is dedicated to generating cloudstack compat payloads, I'm not sure if it's wrong to begin with. The approach below works fine too but forces us into enumerating all the known lists

Reply:
I see your point. My take is that the `query-args` must be somewhat similar to what Aleph does to ensure the signature is constructed similarly on both side. It leads to cryptic error otherwise. That being said, it think we could handle lists and serialise them before hitting `query-args` but still in the payload namespace. Let me give it a shot.